public_html/wp-content/themes/campsite-2017/inc/jetpack.php

```diff
@@ -27,6 +28,19 @@ function jetpack_setup() {
 		)
 	);
 
+	$featured_image_enabled = ! ( function_exists( 'wcorg_skip_feature' ) && wcorg_skip_feature( 'cs17_display_featured_image' ) );
```

Comment:
In https://meta.trac.wordpress.org/ticket/4339#comment:5 we discussed using 2 separate feature flag such that:
1. Change default values for content option so that its disabled by default for old sites, and enabled for new sites.
2. Always enable for new sites

While implementing I realized that 2nd feature flag is probably redundant, because in first one, if we disable by default for old sites it will also automatically enables for new sites, which was the purpose of 2nd feature flag.

What do you think?

Reply:
The reason for two separate flags is that the first one applies to the default settings for all of the places that Jetpack's content options apply (archives, posts, pages), while the second one is specifically for custom post types other than posts and pages (speakers, sponsors, sessions, etc.)

In the notes on that Trac ticket, it questions whether Jetpack's content options for "posts" applies to custom post types as well, or just to the `post` type itself. I think we need to find the answer to that question before this can be resolved.

Reply:
On another note, I think you can omit the `function_exists( 'wcorg_skip_feature' )` part. If that function isn't available, we've got bigger problems ;P

Reply:
Ah got it, I was sure I was missing something. Whether Jetpack's content option will apply to all the post types or not is actually configurable. Right now its onIy enabled for `posts` which means this code actually has a bug and will not fix 3645. 

By configurable, its actually not a Jetpack option, but can be configured in theme like so: https://github.com/WordPress/wordcamp.org/pull/111/files#r280874548 . Right now, CampSite 2017 do not display featured images irrespective of the post type, even when its `post`. 
Which probably also means we may not need different featured flags for different post types, because currently featured image are disabled for all posts? What do you think?

Reply:
Oh, I think I see what you're saying. There's no need to differentiate between regular posts and custom post types because for new sites the same Jetpack content option would apply to them all anyway. Now I'm not sure why we thought we needed two separate feature flags. Any thoughts on this, @iandunn ?

Reply:
`function_exists( 'wcorg_skip_feature' )` removed in https://github.com/WordPress/wordcamp.org/pull/111/commits/e539b28421d0f44e708b3fd945c0bf76df97b08b

Reply:
🤔 I've re-read things a few times, and am still a bit confused, but I think it'd be fine to just have one flag.

It seems like the following is a comprehensive description of what we want to achieve and avoid with the patch:

* New sites should display featured images for all post types, unless the organizer disables that via JP's content options
* Old sites should continue hiding featured images for all post types, unless the organizer enables them via JP's content options

If that's correct, then yeah, I think the current PR achieves that, as long as Corey's suggestion to remove the `! is_single() || is_singular( 'post' ) )` condition is integrated.

Reply:
Feedback addressed in https://github.com/WordPress/wordcamp.org/pull/111/commits/80ec42ce295c21af1f56a52a6715f45fbfb2bf5b . Resolving this conversation. Please unresolve if we should discuss anything further.

Reply:
👍 

FYI @ryelle, in case you notice anything unexpected once we deploy this.